src/kontrol/prove.py

```diff
@@ -468,9 +469,22 @@ def _method_to_cfg(
                 f'Initial state proof {setup_proof.id} for {contract.name_with_path}.{method.signature} still has pending branches.'
             )
 
-        init_node_id = setup_proof.init
+        if setup_proof.bounded:
+            raise RuntimeError(
+                f'Initial state proof {setup_proof.id} for {contract.name_with_path}.{method.signature} still has bounded branches.'
```

Comment:
Does `bounded` indicates that at least of the nodes in the proof is bounded? Do we not allow those in `setUp`?

Reply:
I don't think we should, this means that `setUp` is looping and we haven't gotten out of the loop. I believe `setUp` must fully terminate successfully for us to proceed with the test?

Reply:
That makes sense! I'm just not sure what should happen if we're using BMC — if we've bounding all loops to, say, 3 iterations, that should probably include `setUp` loops and allow continuing the execution after that? Or would we not have `bounded` nodes in this case?

Reply:
Ok, let's discuss this, because I'm not sure what kind of loop we could run into in `setUp`.